examples/ecs/generic_systems.rs

```diff
@@ -0,0 +1,26 @@
+use bevy::prelude::*;
+use bevy::ecs::component::Component;
+
+struct A;
+struct B;
+
+fn spawn_entities_on_click<T: Component, const N: usize>(
+    mut cmds: Commands,
+    mouse_button_inputs: Res<Input<MouseButton>>,
+) {
+    if mouse_button_inputs.just_pressed(MouseButton::Left) {
+        for _ in 0..N {
+            cmds.spawn().insert(T).id();
+        }
+        info!("spawned {} entities with component {}", N, std::any::type_name::<T>());
+    }
+}
```

Comment:
You might need to constrain `T` on being default.

so something like:
```rust
fn spawn_entities_on_click<T, const N: usize>(
    mut cmds: Commands,
    mouse_button_inputs: Res<Input<MouseButton>>,
)
where 
    T: Component + Default
```

and then when inserting
```rust
cmds.spawn().insert(T::default()).id();
```